### ISSUE TITLE
Drag & Drop using Mouse or touch

### ISSUE THREAD

**Issue by MrMojo86** [sschmid/Entitas-CSharp#245]
Hi,

I am bloody newbie to ECS but I want to rewrite my Unity prototype using Entitas which seems to be a good architecture for my project.

In my game I want to drag & drop entities using touch commands. I found no example using drag commands and I am not sure what would be the best way to implement this behaviour in Entitas...

Can someone please help me with this issue?

**Comment by KumoKairo** [comment]
You can't really get away from touch commands like "Input.GetTouch(0)" cause that how Unity works. What you can do, however, is to look at the Physics example in the Entitas Shmup example, which adds some physics collision input to some entity. In case of drag-n-drop however you can basically adds an input component to some entity, say DragComponent with some position and some kind of an ID or reference to a draggable object. Then, every touch move you replace this component with a new one with new position and use some system to update the "Draggable" object.
To recap - 
- Listen to touches in some Unity-aware script that has Update method, 
- Check whether you press on something draggable and create an Entity with a DragComponent (or use some singleton Entity, it's up to you really).
- Write a system that processes those entities (or entity) with the DragComponent to achieve the behaviour you want
